### PR TITLE
8299825: Move StdoutLog and StderrLog to LogConfiguration

### PR DESCRIPTION
**Pull request by jdksjolen**
Hi,

Please consider this small cleanup. `LogConfiguration` initializes and handles the lifetime of `StdoutLog` and `StderrLog`, therefore they should logically belong to that class. I leave the visibility as public for the time being, to change it to private would require the gtests to all be part of a fixture.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8299825](https://bugs.openjdk.org/browse/JDK-8299825): Move StdoutLog and StderrLog to LogConfiguration (**Enhancement** - P4)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Afshin Zafari](https://openjdk.org/census#azafari) (@afshin-zafari - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14376/head:pull/14376` \
`$ git checkout pull/14376`

Update a local copy of the PR: \
`$ git checkout pull/14376` \
`$ git pull https://git.openjdk.org/jdk.git pull/14376/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14376`

View PR using the GUI difftool: \
`$ git pr show -t 14376`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14376.diff">https://git.openjdk.org/jdk/pull/14376.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14376#issuecomment-1582452992)

### CODE REVIEW

File: src/hotspot/share/logging/logConfiguration.cpp [hunk 1]
```diff
@@ -102,20 +105,20 @@ void LogConfiguration::post_initialize() {
 }
 
 void LogConfiguration::initialize(jlong vm_start_time) {
-  StdoutLog = new LogStdoutOutput();
-  StderrLog = new LogStderrOutput();
+  LogConfiguration::StdoutLog = new LogStdoutOutput();
+  LogConfiguration::StderrLog = new LogStderrOutput();
```
Comment (dholmes-ora) [inline src/hotspot/share/logging/logConfiguration.cpp:109]: Style nit: we're in a LogConfiguration method so we don't need to specify it again do we?

File: src/hotspot/share/logging/logConfiguration.cpp [hunk 2]
```diff
@@ -43,6 +43,9 @@
 LogOutput** LogConfiguration::_outputs = nullptr;
 size_t      LogConfiguration::_n_outputs = 0;
 
+LogStdoutOutput* LogConfiguration::LogConfiguration::StdoutLog = nullptr;
+LogStderrOutput* LogConfiguration::LogConfiguration::StderrLog = nullptr;
```
[the remark this answers is not in the document]
Comment (dholmes-ora) [reply src/hotspot/share/logging/logConfiguration.cpp:47]: I'm going to guess after you made the initial change you did a global replace that added `LogConfiguration::` to every occurrence of the names - many of which are not needed.

File: src/hotspot/share/logging/logFileStreamOutput.cpp [hunk 3]
```diff
@@ -32,8 +32,6 @@
 #include "utilities/defaultStream.hpp"
 
 const char* const LogFileStreamOutput::FoldMultilinesOptionKey = "foldmultilines";
-LogStdoutOutput* StdoutLog = nullptr;
-LogStderrOutput* StderrLog = nullptr;
```
Comment (dholmes-ora) [inline src/hotspot/share/logging/logFileStreamOutput.cpp:36]: Surely these had to be declared in a header file somewhere, but I don't see it being removed from a header file,

File: src/hotspot/share/logging/logConfiguration.cpp [hunk 4]
```diff
@@ -421,12 +424,12 @@ bool LogConfiguration::parse_command_line_arguments(const char* opts) {
 
   // Normally options can't be used to change an existing output
   // (parse_log_arguments() will report an error), but we make an exception for
-  // both StdoutLog and StderrLog as they're initialized automatically
+  // both LogConfiguration::StdoutLog and LogConfiguration::StderrLog as they're initialized automatically
```
[the remark this answers is not in the document]
Comment (jdksjolen) [reply src/hotspot/share/logging/logConfiguration.cpp]: OK, removed.

File: src/hotspot/share/logging/logFileStreamOutput.cpp (hunk 3 above)
Comment (jdksjolen) [reply src/hotspot/share/logging/logFileStreamOutput.cpp:36]: Ouch, good catch, thank you.

File: src/hotspot/share/logging/logConfiguration.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (jdksjolen) [reply src/hotspot/share/logging/logConfiguration.cpp]: Yes, I accidentally applied the search and replace twice on this file, but missed this particular instance of the issue.

**Comment by jdksjolen** [comment]
Hi @dholmes-ora, thank you for the review!

**Review by dholmes-ora — APPROVED** [review]
Updates look good - much simpler.

Thanks.

**Review by afshin-zafari — APPROVED** [review]
Looks good to me.

**Comment by jdksjolen** [comment]
Passes tier1, thank you for the reviews.

/integrate

**Pull request closed without merging**